app/models/metric.rb

```diff
@@ -4,6 +4,7 @@ class Metric < ApplicationRecord
 
   belongs_to :track, optional: true
   belongs_to :user, optional: true
+  belongs_to :exercise, optional: true
```

Comment:
I tried to add exercise to the hash, but I don't really know how to do that

Reply:
You definitely don't want this - this implies there's a column `exercise_id` on the `metrics` table, which there isn't.

What you need here is to add this on the child classes. e.g. in `app/models/metrics/submit_iteration_metric.rb` you could add a line like:
```ruby
delegate :exercise, to: :iteration
```

Then see below.